src/controllers/group/group.controller.ts

```diff
@@ -1,80 +1,63 @@
 import { NextFunction, Response } from 'express';
-import * as Joi from 'joi';
-
-import { ResponseStatusCodesEnum } from '../../constants';
-import { ErrorHandler } from '../../errors';
+import { regexFilterParams } from '../../helpers/group';
 import { IGroup, IRequestExtended } from '../../interfaces';
 import { groupService } from '../../services';
-import { groupAttendanceValidator, groupFilterValidator } from '../../validators';
 
 class GroupController {
 
-  async getAllGroups(req: IRequestExtended, res: Response, next: NextFunction) {
-    const {
-      pageSize,
-      pageIndex,
-      offset = pageSize * pageIndex,
-      order = '_id',
-      ...filterParams
-    } = req.query;
-    const filterValidity = Joi.validate(filterParams, groupFilterValidator);
-
-    if (filterValidity.error) {
-      return next(new ErrorHandler(ResponseStatusCodesEnum.BAD_REQUEST, filterValidity.error.details[0].message));
+    async getAllGroups(req: IRequestExtended, res: Response, next: NextFunction) {
+        const {
+            pageSize,
+            pageIndex,
+            offset = pageSize * pageIndex,
+            order = '_id',
+            ...filterParams
+        } = req.query;
+
+        const updatedFilterParams = regexFilterParams(filterParams);
+
+        const groups = await groupService.getAllGroups(updatedFilterParams, +pageSize, offset, order);
+        const count = await groupService.getSizeOfAll(updatedFilterParams) as number;
```

Comment:
Maybe use something like this method to the `pageCount` comment

Reply:
In method to the pageCount we make simple calculations, in this case, we got info of database.
You want a method, which will wrap both of them? (getAllGroup, getSizeOfAll)

Reply:
Discussed in telegram